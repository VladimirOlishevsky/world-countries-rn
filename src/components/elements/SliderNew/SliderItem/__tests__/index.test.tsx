import React from 'react';
import { SlideItem } from '..';
import { render, fireEvent } from '@testing-library/react-native';


const mockedNavigate = jest.fn();
const mockedFetch = jest.fn();

const mockData = {
  item: {
    title: 'Bali, Indonesia',
    img: 'la la la',
    iso_2cc: '22',
  },
  index: 1,
}

const mockStore = {
  countryStore: {
    fetchCountryByCode: mockedFetch
  }
}

jest.mock('store/index', () => ({
  getRootStore: () => mockStore,
}));

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});
describe('SlideItem', () => {

  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('render', () => {
    const tree = render(<SlideItem {...mockData} />);
    expect(tree).toMatchSnapshot()
  });
  test('method navigate', async () => {
    const { findByTestId } = render(<SlideItem {...mockData} />);
    const button = await findByTestId('slideItem')
    expect(mockedNavigate).not.toBeCalled();
    fireEvent(button, 'press');
    expect(mockedNavigate).toHaveBeenCalledTimes(1);
  });
  test('method fetch', async () => {
    const { findByTestId } = render(<SlideItem {...mockData} />);
    const button = await findByTestId('slideItem')
    expect(mockStore.countryStore.fetchCountryByCode).not.toBeCalled();
    fireEvent(button, 'press');
    expect(mockStore.countryStore.fetchCountryByCode).toHaveBeenCalledTimes(1);
  });
});
