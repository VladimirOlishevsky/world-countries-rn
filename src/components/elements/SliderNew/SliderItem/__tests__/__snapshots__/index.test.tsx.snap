// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideItem render 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="slideItem"
>
  <View>
    <Image
      source={
        Object {
          "uri": "la la la",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "height": 368,
          "justifyContent": "center",
          "width": 525,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": undefined,
          "textAlign": "center",
        }
      }
    >
      Bali, Indonesia
    </Text>
  </View>
</View>
`;
